### PR TITLE
Finish clustername removal

### PR DESCRIPTION
**Pull request by lavalamp**
This completes the work started in #108717. It should not merge until merges reopen for 1.25.

The compatibility test will fail until I run the command to update it, which I can't do until the tag for 1.24.0 exists.

```release-note
The `metadata.clusterName` field is completely removed. This should not have any user-visible impact.
```

### CODE REVIEW

**Comment by lavalamp** [comment]
/assign @liggitt @deads2k

I think this is ready now

**Comment by liggitt** [comment]
hold for https://github.com/kubernetes/kubernetes/pull/109803 and rebase on that (will need to regen the API testdata changes)

Since this is a pretty special case (dropping a metadata field we never allowed persisting data in), I think I'd accept stomping the historical fixtures in place, rather than creating "...after_roundtrip" files for all of them. Can do that by temporarily modifying the compatibility test to write back out to the same files when you set UPDATE_COMPATIBILITY_FIXTURE_DATA=true:

```diff
diff --git a/staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/compatibility.go b/staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/compatibility.go
index 8272f9fa358..e2c96f66db4 100644
--- a/staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/compatibility.go
+++ b/staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/compatibility.go
@@ -460,13 +460,13 @@ func (c *CompatibilityTestOptions) runPreviousVersionTest(t *testing.T, gvk sche
 		const updateEnvVar = "UPDATE_COMPATIBILITY_FIXTURE_DATA"
 		if os.Getenv(updateEnvVar) == "true" {
 			if jsonNeedsUpdate {
-				writeFile(t, previousVersionDir, gvk, ".after_roundtrip", "json", jsonAfterRoundTrip)
+				writeFile(t, previousVersionDir, gvk, "", "json", jsonAfterRoundTrip)
 			}
 			if yamlNeedsUpdate {
-				writeFile(t, previousVersionDir, gvk, ".after_roundtrip", "yaml", yamlAfterRoundTrip)
+				writeFile(t, previousVersionDir, gvk, "", "yaml", yamlAfterRoundTrip)
 			}
 			if protoNeedsUpdate {
-				writeFile(t, previousVersionDir, gvk, ".after_roundtrip", "pb", protoAfterRoundTrip)
+				writeFile(t, previousVersionDir, gvk, "", "pb", protoAfterRoundTrip)
 			}
 			t.Logf("wrote expected compatibility data... verify, commit, and rerun tests")
 		} else {
```

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -254,14 +254,8 @@ type ObjectMeta struct {
...
-	// Deprecated: ClusterName is a legacy field that was always cleared by
-	// the system and never used; it will be removed completely in 1.25.
-	//
-	// The name in the go struct is changed to help clients detect
-	// accidental use.
-	//
-	// +optional
-	ZZZ_DeprecatedClusterName string `json:"clusterName,omitempty" protobuf:"bytes,15,opt,name=clusterName"`
+	// Tombstone: ClusterName was a legacy field that was always cleared by
+	// the system and never used. It consumed proto tag 15.
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go:258]: Include the json field name as well. In other places, we actually kept the full field declaration, just commented out... dunno if we want to do that here.

```
	// IPFamily is tombstoned to show why 15 is a reserved protobuf tag.
	// IPFamily *IPFamily `json:"ipFamily,omitempty" protobuf:"bytes,15,opt,name=ipFamily,Configcasttype=IPFamily"`
```

**Comment by liggitt** [comment]
removal lgtm, one comment on the tombstone

**Comment by lavalamp** [comment]
ok, updated...

**Comment by liggitt** [comment]
checked the diff on the fixture files was dropping tag 15:

```diff
--- /dev/fd/63	2022-05-05 16:03:00.000000000 -0400
+++ /dev/fd/62	2022-05-05 16:03:00.000000000 -0400
@@ -37,7 +37,6 @@
       7: 1
     }
     14: "14"
-    15: "15"
     17 {
       1: "16"
       2: "\351\220\212\345\224\212\351\243\231\305\236-U\345\234\264\303\267a/\311\224}\346\221\201(\346\271\227\304\206]"
```

looks like an extensions/v1beta1 NetworkPolicy type got missed in the fixture update? (unit test failure)

**Comment by liggitt** [comment]
/approve

lgtm once the unit test is fixed

**Comment by lavalamp** [comment]
That's very weird, I'll run it again...

**Comment by lavalamp** [comment]
OK, figured it out.

**Comment by lavalamp** [comment]
Actually, wait, I probably have clobbered the base files and should double check that

**Comment by lavalamp** [comment]
OK I think everything should be right now.

**Pull request merged**